clients/src/main/java/org/apache/kafka/common/record/KafkaLZ4BlockInputStream.java

```diff
@@ -142,22 +168,19 @@ private void readBlock() throws IOException {
 
         // verify checksum
         if (flg.isBlockChecksumSet() && Utils.readUnsignedIntLE(in) != checksum.hash(bufferToRead, 0, blockSize, 0)) {
-            throw new IOException(BLOCK_HASH_MISMATCH);
+            throw new LZ4Exception(BLOCK_HASH_MISMATCH);
```

Comment:
The ultimate goal should be to merge all of this code into the upstream lz4 library (lz4-java). Once that happens you'll need to catch the LZ4Exception errors that it raises. Much of this code is written such that it could be merged upstream and we would not have to maintain a separate LZ4 implementation in the kafka codebase.